### PR TITLE
Migrating sequalize

### PR DESCRIPTION
**Pull request by hezhongq**
### Description:
This Pull Request introduces several changes to our register and login endpoints by aligning them with Sequelize ORM for better maintainability and consistency. Additionally, it includes updates to our testing strategy, ensuring that our tests accurately reflect these changes.

### Key Changes:

- Register and Login Endpoint Refactoring:
    - Updated register.js and login.js to use Sequelize for database interactions. This change simplifies the code and leverages the ORM's features for more reliable database operations.
    - Removed direct SQL queries and replaced them with Sequelize's model methods for improved readability and security.
    - Adjusted the login logic to handle usernames in a case-sensitive manner, enhancing data consistency across the application.

- Test Case Updates
    - Modified tc01_register.js and tc02_login.js test cases to align with the updated endpoint logic. These changes ensure that our tests remain relevant and effective in verifying the new Sequelize-based logic.
    - Ensured that test cases handle username case sensitivity correctly, reflecting the changes made in the endpoints.
    - Sequelize Model Testing:

- New npm command to run test
    - `npm test XX` where XX is the test number
    - Ex. `npm test 01` -> this will first seed the data for test case 01 if there is any, then run the test case number 01

- Introduced new tests for Sequelize models to validate their structure and behavior. This addition strengthens our testing suite by verifying the ORM layer directly.

- Seeder File Adjustments:

- Created or updated seeder files (e.g., tc02_seeder.js) to seed test data into the database in a manner consistent with the updated application logic. This ensures that our tests have the necessary data setup for accurate execution.

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -85,8 +85,11 @@ app.use((err, req, res, next) => {
         res.status(500).json({ message: "Unknown error. Please check error log." });
     }
 })
+// // Load the appropriate database configuration based on the environment
+// const env = process.env.NODE_ENV || "development"; // Set a default environment
+// const dbConfig = require("./sequelize.config")[env];
 
-const port = process.env.PORT || 3000;
+const port = process.env.PORT || 3001;
```
Comment (Kianoosh76) [inline backend/app.js:92]: what was the reason to change the port? Do we still need it?

File: backend/models/mark.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      Mark.belongsTo(User, { foreignKey: 'username', as: 'user' });
+      Mark.belongsTo(Task, { foreignKey: 'task', as: 'task' });
+    }
+  }
+  mark.init({
+    mark: {
+      type: DataTypes.NUMERIC,
+      allowNull: false,
+    },
+    oldMark: {
```
Comment (Kianoosh76) [inline backend/models/mark.js:23]: let's keep all column names camel case (old_mark)

File: backend/models_temp/Course.js
Comment (Kianoosh76) [inline backend/models_temp/Course.js:1]: What is the purpose of this folder? Did you mean it to be commited?

File: backend/module/auth/general/login.js
Comment (Kianoosh76) [inline backend/module/auth/general/login.js:1]: Awesome work in reducing the size of this file!

File: backend/module/group/staff/check.js [hunk 3]
```diff
@@ -21,7 +21,83 @@ router.get("/", (req, res) => {
 					} else {
 						if (pg_res_url.rowCount <= 0) {
 							res.status(400).json({ message: "The group id is not found in the database." });
-						} else {
+							const express = require("express");
```
Comment (Kianoosh76) [inline backend/module/group/staff/check.js:24]: why did you add all these raw queries?

File: backend/runTest.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            console.error(`Stderr: ${stderr}`);
+            return;
+        }
+        console.log(`Stdout: ${stdout}`);
+    });
+};
+
+
+// Run migrations, seeder, and the test
+const runTest = async () => {
```
Comment (Kianoosh76) [inline backend/runTest.js:54]: Is this file being used? I see that you're including all phases (migrate, seed, run) in one script. 

But it appears that you're not using it yet?

File: backend/setup/db.js [hunk 5]
```diff
@@ -3,9 +3,9 @@ const { Pool } = require("pg");
 
 const client = new Pool({
 	connectionString: process.env.DATABASE_URL,
-	ssl: {
-		rejectUnauthorized: false
-	}
+	// ssl: {
```
Comment (Kianoosh76) [inline backend/setup/db.js:6]: we should try to avoid leaving commented out code in the codebase.

If these lines should not be there, please remove them

File: backend/tests/run_tests.py
Comment (Kianoosh76) [inline backend/tests/run_tests.py:1]: Is this file being used? It's in python and doesn't run any node command

File: backend/tests/student/__init__.py
Comment (Kianoosh76) [inline backend/tests/student/__init__.py:1]: do we need this file?

File: backend/tests/student/student_test.js
Comment (Kianoosh76) [inline backend/tests/student/student_test.js:1]: do we need this file?

File: backend/tests/ta/ta_test.js
Comment (Kianoosh76) [inline backend/tests/ta/ta_test.js:1]: do we need this file?

File: backend/tests/tc01_registration.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require("../app"); // Adjust the path as per your project structure
+const sequelize = require('../helpers/database'); // Adjust the path as per your project structure
+chai.use(chaiHttp);
+const expect = chai.expect;
+
+describe('Registration Endpoint', () => {
+    beforeEach(async () => {
+        // await truncate();
```
Comment (Kianoosh76) [inline backend/tests/tc01_registration.js:11]: I see that sequelize also attempts to clear and migrate database inside the tests...
Have you tried uncommenting these lines to see it they work? It might be easier than having different node commands

File: postgres/db.sql [hunk 7]
```diff
@@ -1,6 +1,6 @@
 CREATE EXTENSION IF NOT EXISTS pgcrypto;
 
-CREATE TABLE user_info
```
Comment (Kianoosh76) [inline postgres/db.sql:3]: Is this change needed?

**Review by Kianoosh76 — COMMENTED** [review]
Awesome work in setting up the testing infrastructure! We're on a great shape to keep adding stuff to it!

I have a handful of comments that are mostly minor stuff. Once fixed, I'll approve the PR

File: backend/module/group/staff/check.js (hunk 3 above)
Comment (hezhongq) [reply backend/module/group/staff/check.js:24]: It was just copy+paste mistakes..

File: backend/runTest.js (hunk 4 above)
Comment (hezhongq) [reply backend/runTest.js]: Updated runTest.js as it now used for executing the scripts in order:
1. migrateundo: reverte the migrations
2. migrate: apply the migrations
3. seed: seed data to database
4. execute: run the test case based on the test case number that's given

File: backend/tests/tc01_registration.js (hunk 6 above)
Comment (hezhongq) [reply backend/tests/tc01_registration.js:11]: I have attempted to use the sequalize's "beforeEach" and "after" commands and they seems not working.
However, i came across to a solution of using script to help doing the test case setup, so we could remove the body of beforeEach and after for now. I still leave the call with empty body there for future reference in case we need to use them.

File: backend/tests/ta/ta_test.js
Comment (hezhongq) [reply backend/tests/ta/ta_test.js:1]: Not anymore, the original plan was to create a test suite for each model. However, since we are still in progress of creating the test cases while migrating the sequalized endpoints into the backend, it would be good enough to just keep all the test files right under the /tests/ folder, and we could further categorize them in the future.

File: postgres/db.sql (hunk 7 above)
Comment (hezhongq) [reply postgres/db.sql:3]: Nope, seems that it's redundant. I have reverted the change :)

**Review by Kianoosh76 — APPROVED** [review]
